### PR TITLE
🎉 New Source: Opentable Sync API

### PR DESCRIPTION
**Pull request by MaxwellJK**
## What
It is a new source, pulling data from Opentable Sync API.

## How
The solution is based on the HTTP Python CDK solution.

## Recommended reading order
1. `source.py`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@MaxwellJK sorry the long delay, can you share the acceptance test output too?

**Comment by MaxwellJK** [comment]
hey,
if I try to run the acceptance test, one of them fails because of the structure of the json schema...but I can't see the issue!

part of the json schema causing issues:
`    "phone_numbers": {
      "type": ["array", "null"],
      "items": [
        {
          "type": "object",
          "properties": {
            "label": {
              "type": ["string", "null"]
            },
            "number": {
              "type": ["string", "null"]
            },
            "country_code": {
              "type": ["string", "null"]
            },
            "primary": {
              "type": ["boolean", "null"]
            }
          }
        }
      ]
    },`

example of json from the server:
`"phone_numbers": [{"label": "Mobile", "number": "5432345", "country_code": "99", "primary": true}]`

According to any json schema validator, it should works but when I run the acceptance test, I get this:
```collecting ... 
 test_core.py ✓✓✓✓✓✓✓✓✓✓                                                                                                                       91% █████████▏

――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― TestBasicRead.test_read[inputs0] ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

self = <source_acceptance_test.tests.test_core.TestBasicRead object at 0x7f9210437e50>, connector_config = SecretDict(******)
configured_catalog = ConfiguredAirbyteCatalog(streams=[ConfiguredAirbyteStream(stream=AirbyteStream(name='guests', json_schema={'$schema': ... cursor_field=['updated_at_utc'], destination_sync_mode=<DestinationSyncMode.append: 'append'>, primary_key=[['id']])])
inputs = BasicReadTestConfig(config_path='secrets/config.json', configured_catalog_path='integration_tests/configured_catalog.json', empty_streams=set(), expect_records=None, validate_schema=True, timeout_seconds=None)
expected_records = [], docker_runner = <source_acceptance_test.utils.connector_runner.ConnectorRunner object at 0x7f921041eb90>
detailed_logger = <AirbyteNativeLogger detailed_logger /test_input/acceptance_tests_logs/test_core.py__TestBasicRead__test_read[inputs0].txt (DEBUG)>

    def test_read(
        self,
        connector_config,
        configured_catalog,
        inputs: BasicReadTestConfig,
        expected_records: List[AirbyteMessage],
        docker_runner: ConnectorRunner,
        detailed_logger,
    ):
        output = docker_runner.call_read(connector_config, configured_catalog)
        records = [message.record for message in filter_output(output, Type.RECORD)]
    
        assert records, "At least one record should be read using provided catalog"
    
        if inputs.validate_schema:
>           self._validate_schema(records=records, configured_catalog=configured_catalog)

source_acceptance_test/tests/test_core.py:262: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
source_acceptance_test/tests/test_core.py:198: in _validate_schema
    TestBasicRead._validate_records_structure(records, configured_catalog)
source_acceptance_test/tests/test_core.py:183: in _validate_records_structure
    schemas[stream.stream.name] = set(get_expected_schema_structure(stream.stream.json_schema))
source_acceptance_test/utils/json_schema_helper.py:211: in get_expected_schema_structure
    _scan_schema(schema)
source_acceptance_test/utils/json_schema_helper.py:205: in _scan_schema
    return {k: _scan_schema(v, path + "/" + k) for k, v in props.items()}
source_acceptance_test/utils/json_schema_helper.py:205: in <dictcomp>
    return {k: _scan_schema(v, path + "/" + k) for k, v in props.items()}
source_acceptance_test/utils/json_schema_helper.py:208: in _scan_schema
    return [_scan_schema(items, path + "/[]")]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

subschema = [{'additionalProperties': {'country_code': {'type': ['string', 'null']}, 'label': {'type': ['string', 'null']}, 'number': {'type': ['string', 'null']}, 'primary': {'type': ['boolean', 'null']}}, 'type': 'object'}]
path = '/phone_numbers/[]'

    def _scan_schema(subschema, path=""):
        if "oneOf" in subschema or "anyOf" in subschema:
            if annotate_one_of:
                return [
                    _scan_schema({"type": "object", **s}, path + f"({num})")
                    for num, s in enumerate(subschema.get("oneOf") or subschema.get("anyOf"))
                ]
            return [_scan_schema({"type": "object", **s}, path) for s in subschema.get("oneOf") or subschema.get("anyOf")]
>       schema_type = subschema.get("type", ["null"])
E       AttributeError: 'list' object has no attribute 'get'

source_acceptance_test/utils/json_schema_helper.py:194: AttributeError
------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------
{"type": "LOG", "log": {"level": "WARN", "message": "Unable to parse connector's output 1 validation error for AirbyteMessage\n__root__\n  Extra data: line 1 column 4 (char 3) (type=value_error.jsondecode; msg=Extra data; doc=319dc4c5-3170-4528-8b93-4d721a177958 1519276; pos=3; lineno=1; colno=4)"}}
{"type": "LOG", "log": {"level": "WARN", "message": "Unable to parse connector's output 1 validation error for AirbyteMessage\n__root__\n  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=Calling endpoint https://platform.opentable.com/sync/v2/guests?offset=0&limit=1000&rid=730843&updated_after=2021-11-08T20%3A49%3A31Z; pos=0; lineno=1; colno=1)"}}
{"type": "LOG", "log": {"level": "WARN", "message": "Unable to parse connector's output 1 validation error for AirbyteMessage\n__root__\n  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=Calling endpoint https://platform.opentable.com/sync/v2/guests?offset=0&limit=1000&rid=29437&updated_after=2021-11-08T20%3A49%3A31Z; pos=0; lineno=1; colno=1)"}}
{"type": "LOG", "log": {"level": "WARN", "message": "Unable to parse connector's output 1 validation error for AirbyteMessage\n__root__\n  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=Calling endpoint https://platform.opentable.com/sync/v2/reservations?offset=0&limit=1000&rid=730843&updated_after=2021-11-08T20%3A49%3A31Z; pos=0; lineno=1; colno=1)"}}
{"type": "LOG", "log": {"level": "WARN", "message": "Unable to parse connector's output 1 validation error for AirbyteMessage\n__root__\n  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=Calling endpoint https://platform.opentable.com/sync/v2/reservations?offset=0&limit=1000&rid=29437&updated_after=2021-11-08T20%3A49%3A31Z; pos=0; lineno=1; colno=1)"}}
--------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------
WARNING  root:connector_runner.py:112 Unable to parse connector's output 1 validation error for AirbyteMessage
__root__
  Extra data: line 1 column 4 (char 3) (type=value_error.jsondecode; msg=Extra data; doc=319dc4c5-3170-4528-8b93-4d721a177958 1519276; pos=3; lineno=1; colno=4)
WARNING  root:connector_runner.py:112 Unable to parse connector's output 1 validation error for AirbyteMessage
__root__
  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=Calling endpoint https://platform.opentable.com/sync/v2/guests?offset=0&limit=1000&rid=730843&updated_after=2021-11-08T20%3A49%3A31Z; pos=0; lineno=1; colno=1)
WARNING  root:connector_runner.py:112 Unable to parse connector's output 1 validation error for AirbyteMessage
__root__
  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=Calling endpoint https://platform.opentable.com/sync/v2/guests?offset=0&limit=1000&rid=29437&updated_after=2021-11-08T20%3A49%3A31Z; pos=0; lineno=1; colno=1)
WARNING  root:connector_runner.py:112 Unable to parse connector's output 1 validation error for AirbyteMessage
__root__
  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=Calling endpoint https://platform.opentable.com/sync/v2/reservations?offset=0&limit=1000&rid=730843&updated_after=2021-11-08T20%3A49%3A31Z; pos=0; lineno=1; colno=1)
WARNING  root:connector_runner.py:112 Unable to parse connector's output 1 validation error for AirbyteMessage
__root__
  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=Calling endpoint https://platform.opentable.com/sync/v2/reservations?offset=0&limit=1000&rid=29437&updated_after=2021-11-08T20%3A49%3A31Z; pos=0; lineno=1; colno=1)

 test_core.py ⨯                                                                                                                               100% ██████████
================================================================== short test summary info ==================================================================
FAILED ../../test_input/test_core.py::TestBasicRead::test_read[inputs0] - AttributeError: 'list' object has no attribute 'get'

Results (41.44s):
      10 passed
       1 failed
         - ../airbyte/source_acceptance_test/source_acceptance_test/tests/test_core.py:247 TestBasicRead.test_read[inputs0]

File: airbyte-integrations/connectors/source-opentable-sync-api/integration_tests/catalog.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        "properties": {
+          "column1": {
+            "type": "string"
+          },
+          "column2": {
+            "type": "number"
+          }
+        }
+      }
+    },
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-opentable-sync-api/integration_tests/catalog.json:20]: can we remove this if we are not using it

File: airbyte-integrations/connectors/source-opentable-sync-api/integration_tests/invalid_config.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "todo-wrong-field": "this should be an incomplete config file, used in standard tests"
+}
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-opentable-sync-api/integration_tests/invalid_config.json:3]: ^ same can we remove this if we are not using

**Comment by marcosmarxm** [comment]
Hello @MaxwellJK ! Sorry to not finished your contribution before the date stipulated in the contest. All contributions made before 15-November are eligible to receive the award. We're trying to review your contribution as soon as possible.

**Comment by karinakuz** [comment]
I've submitted a request to become an OpenTable Partner.
If we are approved they will be contacting us within 3-4 weeks.

**Comment by marcosmarxm** [comment]
@MaxwellJK I added the label blocked. Let me know if you need any help or want to continue working in this connector.

**Pull request closed without merging**